addons/mrp/tests/test_order.py

```diff
@@ -512,7 +512,7 @@ def test_product_produce_2(self):
 
         self.assertEqual(len(product_produce.workorder_line_ids), 3, 'You should have 3 produce lines. One for each serial to consume')
         product_produce.qty_producing = 1
-        produce_line_1 = product_produce.workorder_line_ids[0]
+        produce_line_1 = product_produce.workorder_line_ids.filtered(lambda line: line.product_id == p1)[0]
```

Comment:
this change may make sense but i don't get where the change in order comes from.. please explain